### PR TITLE
Martial arts now use a weight system to determine which one is active

### LINKED ISSUE #13501: You can wear krav maga gloves to turn off sleeping carp and shoot guns
**Issue Description**:

By putting on krav maga gloves, you overide sleeping carp, allowing you to shoot guns

**What did you expect to happen**: 

Gaining the power of sleeping carp and krav maga

**What happened instead**:

Sleeping carp is turned  off, you can be shot, BUT you can also shoot guns as well

**Why is this bad/What are the consequences**:

Sleeping carp is not something you should be able to turn on and off at will to shoot guns.

**Steps to reproduce the problem**:

Spawn and use a sleeping carp scroll
Try and fail to shoot a gun
Put on krav maga gloves
Notice you are now able to shoot guns, and to be shot with ranged weapons

**When did the problem start happening**:

Probably the martial art rework

### PR DESCRIPTION
**Pull request by BiancaWilkson**
## What Does This PR Do

This PR changes how martial arts are stored, and gives each one a weight for determining which ones should be overridden by others.
## Why It's Good For The Game

Right now, getting new martial arts does not properly remove the old one, resulting in a weird mess of having multiple instances of CQC or carp. Not only does this fix some runtimes, but current martial arts are automatically overriden by any new ones you get. This means that getting drunk, or putting on boxing gloves will temporarily disable any other martial arts someone might have. This system makes it so they learn the new martial art, but if it's a lower weight value than their current one, it won't be as the active martial art.

Fixes #13501
## Testing

Spawned in, gave myself many duplicate martial arts, gave myself many different martial arts and made sure they were weighted properly, and tested chef's CQC to make sure that did not break either.
## Changelog
:cl:
tweak: New martial arts don't automatically override old ones, instead using a weighted system to determine what's active.
/:cl:

### CODE REVIEW

File: code/modules/martial_arts/martial.dm [hunk 1]
```diff
@@ -33,6 +33,8 @@
 	var/in_stance = FALSE
 	/// If the martial art allows parrying.
 	var/can_parry = FALSE
+	/// The priority of which martial art is picked from all the ones someone knows.
```
Comment (Sirryan2002) [inline code/modules/martial_arts/martial.dm:36]: can you specifies what the range of values mean i.e. 0 = lowest priority   and X... being the highest priority?

File: code/modules/martial_arts/martial.dm [hunk 2]
```diff
@@ -136,27 +138,29 @@
...
-	H.mind.martial_art = null // Remove reference
+	else
+		H.mind.known_martial_arts.Remove(MA)
+		H.mind.martial_art = get_highest_weight(H)
 	H.verbs -= /mob/living/carbon/human/proc/martial_arts_help
-	if(base)
-		base.teach(H)
-		base = null
+
+/datum/martial_art/proc/get_highest_weight(mob/living/carbon/human/H)
```
Comment (Sirryan2002) [inline code/modules/martial_arts/martial.dm:158]: can you document this proc please?

**Comment by SteelSlayer** [comment]
@BiancaWilkson I heard an in-game bug report of someone saying they had 3 instance of the "defensive stance" action button. Seems like CQC's `teach()` is being called multiple times? Would this PR fix that issue?

File: code/modules/martial_arts/krav_maga.dm [hunk 3]
```diff
@@ -24,12 +25,16 @@
 	button_icon_state = "neckchop"
 
 /datum/action/neck_chop/Trigger()
+	var/mob/living/carbon/human/H = owner
+	var/datum/martial_art/krav_maga/krav = new()
+	if(!istype(krav, H.mind.martial_art))
```
[the remark this answers is not in the document]
Comment (SteelSlayer) [reply code/modules/martial_arts/krav_maga.dm]: Have you tried `istype(H.mind.martial_art, /datum/martial_art/krav_maga)`? Should work.

File: code/modules/martial_arts/martial.dm [hunk 4]
```diff
@@ -136,27 +138,30 @@
...
 /datum/martial_art/proc/remove(mob/living/carbon/human/H)
+	var/datum/martial_art/MA = src
 	if(!H.mind)
 		return
-	if(H.mind.martial_art != src)
-		return
-	H.mind.martial_art = null // Remove reference
+	else
+		H.mind.known_martial_arts.Remove(MA)
+		H.mind.martial_art = get_highest_weight(H)
```
Comment (Sirryan2002) [inline code/modules/martial_arts/martial.dm:155]:
```suggestion
	H.mind.known_martial_arts.Remove(MA)
	H.mind.martial_art = get_highest_weight(H)
```

**Review by Sirryan2002 — APPROVED** [review]

**Review by S34NW — DISMISSED** [review]
Please implement sirryan's changes

**Review by Sirryan2002 — APPROVED** [review]

**Review by SteelSlayer — APPROVED** [review]

**Pull request merged**